sdk/identity/identity/src/credentials/clientCertificateCredential.ts

```diff
@@ -59,135 +33,32 @@ export class ClientCertificateCredential implements TokenCredential {
+   * @param options - Optional parameters. See {@link GetTokenOptions}.
    */
-  public async getToken(
+  async getToken(
     scopes: string | string[],
-    options?: GetTokenOptions
+    options: GetTokenOptions = {}
   ): Promise<AccessToken | null> {
-    const { span, updatedOptions } = createSpan("ClientCertificateCredential-getToken", options);
-    try {
-      const tokenId = uuidV4();
-      const urlSuffix = getIdentityTokenEndpointSuffix(this.tenantId);
-      const audienceUrl = `${this.identityClient.authorityHost}/${this.tenantId}/${urlSuffix}`;
-      let header: jws.Header;
-
-      if (this.certificateX5c) {
-        header = {
-          typ: "JWT",
-          alg: "RS256",
-          x5t: this.certificateX5t,
-          x5c: this.certificateX5c
-        };
-      } else {
-        header = {
-          typ: "JWT",
-          alg: "RS256",
-          x5t: this.certificateX5t
-        };
-      }
-
-      const payload = {
-        iss: this.clientId,
-        sub: this.clientId,
-        aud: audienceUrl,
-        jti: tokenId,
-        nbf: timestampInSeconds(new Date()),
-        exp: timestampInSeconds(addMinutes(new Date(), SelfSignedJwtLifetimeMins))
-      };
-
-      const clientAssertion = jws.sign({
-        header,
-        payload,
-        secret: this.certificateString
-      });
-
-      const webResource = this.identityClient.createWebResource({
-        url: audienceUrl,
-        method: "POST",
-        disableJsonStringifyOnBody: true,
-        deserializationMapper: undefined,
-        body: qs.stringify({
-          response_type: "token",
-          grant_type: "client_credentials",
-          client_id: this.clientId,
-          client_assertion_type: "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
-          client_assertion: clientAssertion,
-          scope: typeof scopes === "string" ? scopes : scopes.join(" ")
-        }),
-        headers: {
-          Accept: "application/json",
-          "Content-Type": "application/x-www-form-urlencoded"
-        },
-        abortSignal: options && options.abortSignal,
-        spanOptions: updatedOptions?.tracingOptions?.spanOptions
-      });
-
-      const tokenResponse = await this.identityClient.sendTokenRequest(webResource);
-      logger.getToken.info(formatSuccess(scopes));
-      return (tokenResponse && tokenResponse.accessToken) || null;
-    } catch (err) {
-      const code =
-        err.name === AuthenticationErrorName
-          ? CanonicalCode.UNAUTHENTICATED
-          : CanonicalCode.UNKNOWN;
-      span.setStatus({
-        code,
-        message: err.message
-      });
-      logger.getToken.info(formatError("", err));
-      throw err;
-    } finally {
-      span.end();
-    }
+    return trace("ClientCertificateCredential.getToken", options, async (newOptions) => {
+      const arrayScopes = typeof scopes === "object" ? scopes : [scopes];
+      return this.msalClient.getToken(arrayScopes, newOptions);
```

Comment:
These getToken call to the MSAL client implementation's doGetToken (there's one for DeviceCode, another one for client secret, etc).

Maor inspired me to go in this direction!